spring-pulsar-test/src/test/java/org/springframework/pulsar/test/support/PulsarConsumerTestUtilTests.java

```diff
@@ -97,9 +100,101 @@ void exceptionIsThrownWhenConditionNotMetWithinAwaitDuration() {
 			.withMessage("Condition was not met within 5 seconds");
 	}
 
+	@Test
+	void messagesAreConsumedWhenContainerIsRunningAndConsumeMessagesIsCalledWithoutArguments() {
+		// depends upon pulsarClient created in prepareForTest
+		var topic = testTopic("e1");
+		IntStream.range(0, 2).forEach(i -> pulsarTemplate.send(topic, "message-" + i));
+		var msgs = PulsarConsumerTestUtil.<String>consumeMessages()
+			.fromTopic(topic)
+			.withSchema(Schema.STRING)
+			.awaitAtMost(Duration.ofSeconds(5))
+			.until(desiredMessageCount(2))
+			.get();
+		assertThat(msgs).hasSize(2);
+	}
+
+	@Test
+	void messagesAreConsumedWhenContainerIsStoppedAndConsumeMessagesIsCalledWithoutArguments() {
+		PulsarTestContainerSupport.stopContainer();
+		PulsarConsumerTestUtil.<String>consumeMessages();
+		// TODO: Complete this test
```

Comment:
I'd like to request some assistance writing this test
- Basically I want to test if consumeMessages works properly even if `PULSAR_CONTAINER` is not running
- When I try to `get()` the consumeMessages I get connection refused error for the localhost pulsar client
- Also I'm not sure at what point should I send messages to the topic, if I do it before stopping the container will it still work? since the pulsarClient instance would be different after container is stopped

Reply:
Yep, this one is tricky to test. We actually can't stop the container as the other tests will fail. The "magic" of the PTCS is that it has the `@BeforeAll` that starts it for the test class. If we stopped it we would have to restart it and then it gets into a bit of flaky-test-land at that point. 

I think instead we can static mock the "isContainerStarted" that it uses to decide what to do. 

We are releasing `1.1.0-M2` tomorrow so I am going to take what you have in this current state and add a polish commit on top so that we can get this in. I will include this test adjustment in that commit. 

Great work @KartikShrivastava !

Reply:
Also, if you are wondering why we did not have (or need) the stop container method annotated w/ a `@AfterAll` to match the start w/ / `@BeforeAll` it is because Testcontainers does

> At the end of the test suite the [Ryuk container](https://github.com/testcontainers/moby-ryuk) that is started by Testcontainers core will take care of stopping the singleton container.

However, I think I will add it in the polish commit to avoid future confusion

Reply:
> [!NOTE]
> Adding the `@AfterAll` to the new `stopContainer` method had unintended side-effects when used in a Jupiter `@Nested` test. I removed the annotation but left the method (https://github.com/spring-projects/spring-pulsar/commit/9bc6bda306b347d47b7b7a5d44b4e56db5f81798)

Reply:
Cool, thanks for taking this up @onobc, really appreciate that.
I've one minor question seeing the improvements you've made: Is there a recommended way or methodology we follow to name test methods in this project? If so, then I would like to learn/read more about it :)

Reply:
You are welcome @KartikShrivastava . 
In general we follow the "when-THIS-then-OUTCOME" but more than anything the principle I follow is consistency w/in a test suite and enough information in the name to understand what it is testing. I will also typically group tests together using `@Nested` which groups them and allows us to not have ultra long test names. I was tempted to do some grouping on this test but ended up leaving as is. 

I also noticed that the negative test cases I named a bit different than the other ones (they are shorter in nature). 

So there is no doc that describes this but it would be a good idea for us to add one. I will note this.

Thanks